File: backend/src/main/java/org/kmurygin/healthycarbs/user/dto/UserDTO.java
package org.kmurygin.healthycarbs.user.dto;

import lombok.*;
import org.kmurygin.healthycarbs.user.model.Role;

import java.time.Instant;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {
    private Long id;
    private String username;
    private String firstName;
    private String lastName;
    private String email;
    private Role role;
    private Long profileImageId;
    private Instant createdAt;
    private Instant lastLoginAt;
    private Boolean isActive;
}

File: frontend/src/app/core/models/dto/user.dto.ts
import type {UserRole} from "@core/models/enum/user-role.enum";

export interface UserDto {
  id: number;
  username: string;
  firstName: string;
  lastName: string;
  email: string;
  profileImageId: number | null;
  role: UserRole;
  createdAt: string | null;
  lastLoginAt: string | null;
  isActive: boolean;
}
